capstone/src/test/java/com/google/sps/objects/ChallengeTest.java

```diff
@@ -0,0 +1,134 @@
+package com.google.sps.objects;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.Objects.Badge;
+import com.google.sps.Objects.Challenge;
+
+/**
+ * Unit tests for {@link Challenge}.
+ *
+ * @author lucyqu
+ */
+public class ChallengeTest {
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+          new LocalDatastoreServiceTestConfig()
+              .setDefaultHighRepJobPolicyUnappliedJobPercentage(0));
+
+  private Challenge challenge;
+  private Badge badge;
+  private final String CHALLENGE_NAME = "Run 3 miles";
+  private final long DUE_DATE = 12345;
+  private final long ID = 100;
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+    badge = new Badge(/* challenge name */ CHALLENGE_NAME, /* icon */ null, /* timestamp */ 234567);
+    challenge =
+        new Challenge(
+            CHALLENGE_NAME, /* challenge name */
+            DUE_DATE, /* due date */
+            badge, /* badge */
+            new ArrayList<String>(), /* completed users */
+            ID /* challenge id */);
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+    challenge = null;
```

Comment:
Agreed, no need to null it out on tearDown. I almost never use a teardown unless you really need to destroy/deconstruct something, since the JVM will automatically garbage-collect the original object once your replace the challenge